### ISSUE TITLE
About Code Reproduction and Pretraining Issues

### ISSUE THREAD

**Issue by ConnerWK** [LL3RD/ASA#2]
Dear authors,

Thanks for your great work and I have a few questions about the code:

1. I used the ADNI dataset and the provided pre-training code for pre-training, but I found that the training process was difficult to converge, or in other words, the loss was difficult to decrease to a smaller value (the current code uses the MSE loss function provide in this repo), and it was difficult to obtain good reconstruction results. During pre-training, in addition to using the preprocessing method provided by nnFormer, do I need to map the grayscale range to [0, 1] or [-1, 1]? How can I improve the pre-training process?
2. Has the author visualized the pre-training process? Even when using the pre-training weights provided by this repository to generate reconstructions on masked data, only abstract results can be obtained. I have tried using Z-Score and the data preprocessing method provided by nnFormer, and mapping the grayscale to [0, 1] or [-1, 1], but I have not obtained good results. Could it be a problem with how I am normalizing the original data? Could the author provide some advice or guidance? 
 
Thank you very much!

**Comment by LL3RD** [comment]
Due to the long time interval, I can only give some suggestions and the specific data and follow-up work was assigned to others.
For ADNI dataset processing, we only use the nnFormer/nnUnet preprocessing process and use the data in nnUNet_preprocessed as pre-training data.
And for the pre-training weights provided, whether the weights are completely matched when loading？ If it matches exactly, you can try to use the data preprocessed by nnUnet as input.

**Comment by ConnerWK** [comment]
Thank you for your prompt response.

Yes, I did try the preprocessing methods of nnFormer/nnUNet, but when using the provided weights to generate and visualize the results with a mask rate of 0.75, I only obtained some abstract results. It seems that the provided weights are consistent in size with the pre-trained model weights in the repository, except for the weight of the key 'encoder.vit.patch_embedding.position_embeddings', which has a size of [4096, 384], while the corresponding weight provided in the current code has a size of [1, 4096, 384]. Because the difference doesn't seem significant, I added a dimension to the weight to change the size from [4096, 384] to [1, 4096, 384]. The other weights seem to have the same size.

Did the author visualize the reconstructed results during the code construction process? Or is it only necessary to observe the decrease in the loss curve to judge the quality of pre-training? Because I observed that the loss would decrease during pre-training and the loss function did indeed reach a plateau that was difficult to continue to decrease after 200 epochs (using only MSE loss), and the author chose the weights from the 199th epoch.

Does the current network not necessarily require a good reconstruction result, but rather to learn the overall brain dataset information during pre-training? In other words, the quality of the pre-training reconstruction result may not be the key factor in improving downstream tasks, and it is more important to learn the information from the entire dataset during pre-training.

**Comment by LL3RD** [comment]
In the appendix, we show the visualization pictures of the pre-training stage. in fact, only blurred results are actually reconstructed.
[MAE-Visual.pdf](https://github.com/LL3RD/ASA/files/11061147/MAE-Visual.pdf)

**Comment by ConnerWK** [comment]
Thank you very much. This is quite similar to the results of my local pre-training. Initially, I thought the network had only learned a vague, general brain template. Also, I would like to ask if the visualization results for different cases are similar? In my local pre-training results, they are slightly more blurred than the appendix you provided, and although there are differences in grayscale for different examples, they all look quite similar.

**Comment by LL3RD** [comment]
Indeed, the generated results in different cases are relatively similar. We think the pre-training helps the model learn generalizable representations for brain structures, which is useful for downstream tasks.

**Comment by ConnerWK** [comment]
Thanks very much！
